0.Preprocess_Data/scripts/1.preprocessing_morphology_data.py

```diff
@@ -239,40 +239,75 @@
     "oneb_Metadata_Treatment_Dose_Inhibitor_Dose"
-].str.replace("media_ctr_0.0_0_Media_ctr_0_0", "media_ctr_0.0_0_Media_ctr_0.0_0")
+].str.replace("media_ctr_0.0_0_Media_ctr_0_0", "Media", regex=False)
 
+feature_df["oneb_Metadata_Treatment_Dose_Inhibitor_Dose"] = feature_df[
+    "oneb_Metadata_Treatment_Dose_Inhibitor_Dose"
+].str.replace("media_ctr_0.0_0_Media_0_0", "Media", regex=False)
 
-# In[18]:
+feature_df["oneb_Metadata_Treatment_Dose_Inhibitor_Dose"] = feature_df[
+    "oneb_Metadata_Treatment_Dose_Inhibitor_Dose"
+].str.replace("media_ctr_0.0_0_Media_ctr_0.0_0", "Media", regex=False)
 
+feature_df["oneb_Metadata_Treatment_Dose_Inhibitor_Dose"] = feature_df[
+    "oneb_Metadata_Treatment_Dose_Inhibitor_Dose"
+].str.replace(
+    "Flagellin_1.000_0_Disulfiram_1.000_uM",
+    "Flagellin_1.000_ug_per_ml_Disulfiram_1.000_uM",
+    regex=False,
+)
 
-# need to convert to strings to save as parquet
-# if the column is an object then convert it to a string
-for column in feature_df.columns:
-    if feature_df[column].dtype == "object":
-        feature_df[column] = feature_df[column].astype(str)
+feature_df["oneb_Metadata_Treatment_Dose_Inhibitor_Dose"] = feature_df[
+    "oneb_Metadata_Treatment_Dose_Inhibitor_Dose"
+].str.replace(
+    "Flagellin_1.000_0_DMSO_0.025_%",
+    "Flagellin_1.000_ug_per_ml_DMSO_0.025_%",
+    regex=False,
+)
 
+feature_df["oneb_Metadata_Treatment_Dose_Inhibitor_Dose"] = feature_df[
+    "oneb_Metadata_Treatment_Dose_Inhibitor_Dose"
+].str.replace(
+    "Flagellin_0.100_ug_per_ml_DMSO_0.000_%",
+    "Flagellin_0.100_ug_per_ml_DMSO_0.025_%",
+    regex=False,
+)
 
-# In[19]:
+feature_df["oneb_Metadata_Treatment_Dose_Inhibitor_Dose"] = feature_df[
+    "oneb_Metadata_Treatment_Dose_Inhibitor_Dose"
+].str.replace(
+    "Flagellin_1.000_ug_per_ml_DMSO_0.000_%",
+    "Flagellin_1.000_ug_per_ml_DMSO_0.025_%",
+    regex=False,
+)
 
+feature_df["oneb_Metadata_Treatment_Dose_Inhibitor_Dose"] = feature_df[
+    "oneb_Metadata_Treatment_Dose_Inhibitor_Dose"
+].str.replace("media_ctr_Media_ctr_nan__0.0_µg_per_ml", "Media", regex=False)
 
-# write to parquet file
-feature_df.to_parquet(feature_df_out_path)
+feature_df["oneb_Metadata_Treatment_Dose_Inhibitor_Dose"] = feature_df[
+    "oneb_Metadata_Treatment_Dose_Inhibitor_Dose"
+].str.replace("media_ctr_0.0_ug_per_ml_Media_ctr_0_0", "Media", regex=False)
 
 
-# In[20]:
+# In[18]:
 
 
-del feature_df
-import gc
+print(len(feature_df["oneb_Metadata_Treatment_Dose_Inhibitor_Dose"].unique()))
 
-gc.collect()
 
+# In[19]:
 
-# In[21]:
 
+# need to convert to strings to save as parquet
+# if the column is an object then convert it to a string
+for column in feature_df.columns:
+    if feature_df[column].dtype == "object":
+        feature_df[column] = feature_df[column].astype(str)
 
-# Define inputs
-feature_file = pathlib.Path(f"../data/{cell_type}_sc.parquet")
-feature_df = pd.read_parquet(feature_file)
-# quick check for site
-print(feature_df["Metadata_Site"])
+
+# In[ ]:
```

Comment:
Looks like this code cell wasn't run. Recommend making sure that all cells are run sequentially.